src/main/java/us/kilroyrobotics/subsystems/LEDs.java

```diff
@@ -49,30 +49,30 @@ public static enum LEDMode {
 
     /** Creates a new LEDs. */
     public LEDs() {
-        this.led.setColorOrder(ColorOrder.kBRG);
-        this.led.setLength(this.ledBuffer.getLength());
-        this.led.setData(this.ledBuffer);
-        this.led.start();
+        led.setColorOrder(ColorOrder.kBRG);
+        led.setLength(ledBuffer.getLength());
+        led.setData(ledBuffer);
+        led.start();
     }
 
     public void setMode(LEDMode newMode) {
-        this.mode = newMode;
+        mode = newMode;
     }
 
     @Override
     public void periodic() {
         switch (mode) {
             case Rainbow:
-                this.rainbow.applyTo(ledBuffer);
+                rainbow.applyTo(ledBuffer);
                 break;
             case TeleopAligned:
-                this.teleopAligned.applyTo(ledBuffer);
+                teleopAligned.applyTo(ledBuffer);
                 break;
             case WaitingForCoral:
-                this.waitingForCoral.applyTo(ledBuffer);
+                waitingForCoral.applyTo(ledBuffer);
                 break;
             case CoralDetected:
-                this.coralGrabbed.applyTo(ledBuffer);
+                coralGrabbed.applyTo(ledBuffer);
                 break;
             case Defense:
                 this.defense.applyTo(ledBuffer);
```

Comment:
We should be able to make this an enum where the various modes, i.e. rainbow, are apart of the enum.

So periodic could instead be mode.getPattern.applyTo(ledBuffer).

Or we could add a method to the enum that applies it to the led buffer, so periodic becomes mode.apply()

Reply:
Maybe, but that would be a seperate PR